api/spec/src/billing/invoice/bill.tsp

```diff
@@ -0,0 +1,407 @@
+  @visibility("read", "query") // We only populate this for voided invoices
+  preceding?: Org.DocumentRef;
+
+  @summary("Summary of all the invoice totals, including taxes (calculated).")
+  @visibility("read", "query")
+  totals?: Totals;
+
+  @summary("Unstructured information that is relevant to the invoice, such as correction or additional legal details.")
+  @visibility("read", "query")
+  notes?: CBC.Note[];
+
+  @summary("Additional semi-structured data that doesn’t fit into the body of the invoice.")
+  @visibility("read", "query")
+  meta?: CBC.Meta;
+}
+
+namespace OpenMeter {
+  @friendlyName("BillingInvoiceSummaryOpenMeter")
+  @summary("Invoice summary represents the invoice object's OpenMeter extensions.")
+  model InvoiceSummary {
+    @visibility("read", "query", "update")
+    id: ULID;
+
+    @visibility("read", "query")
+    @summary("The status of the invoice.")
+    status: InvoiceStatus;
+
+    @summary("The time the invoice was issued.")
+    @visibility("read", "query")
+    // NOTE: This replaces the issueDate in the gobl schema, as date only is a bad practice and should be avoided.
+    issuedAt?: DateTime;
+
+    @summary("The workflow settings associated with this invoice")
+    @visibility("read", "query") // Seperate API endpoint is available for updating
+    workflow?: InvoiceWorkflowSettings;
+  }
+
+  @friendlyName("BillingInvoiceWorkflowSettings")
+  model InvoiceWorkflowSettings {
+    /**
+     * updatedAt is the time when the invoice workflow was last updated.
+     *
+     * This field is updated when the invoice workflow is updated, during put requests we
+     * are validating the update time to ensure there is no parallel updates happening.
+     */
+    updatedAt: DateTime;
+
+    @summary("Should the invoice auto advance to the next status?")
+    autoAdvance: boolean;
+
+    /**
+     * The invoice stays in the draft status until this time.
+     *
+     * If auto advance is disabled the value will be empty.
+     */
+    draftUtil?: DateTime;
+
+    /**
+     * The period after which the invoice is due.
+     *
+     * default: 7 days
+     */
+    @encode(DurationKnownEncoding.ISO8601)
+    @example(duration.fromISO("P1D"))
+    dueAfter?: duration;
+  }
+
+  /**
+   * InvoiceStatus describes the status of an invoice.
+   */
+  @friendlyName("BillingInvoiceStatus")
+  enum InvoiceStatus {
+    /**
+     * The list of line items for the next invoice is being gathered.
+     */
+    gathering: "gathering",
+
+    /**
+     * The invoice is waiting for review by our partner.
+     */
+    review: "review",
+
+    /**
+     * The invoice has been issued to the customer.
+     */
+    issued: "issued",
+
+    /**
+     * The invoice has been paid by the customer.
+     */
+    payment_received: "payment_received",
+
+    /**
+     * A manual approval is required before the invoice can be issued.
+     */
+    manual_approval_required: "manual_approval_required",
```

Comment:
What's the format for enums? cc @tothandras @sagikazarmark

Reply:
That's somewhat all over the place. Meters:

```
@friendlyName("WindowSize")
enum WindowSize {
  MINUTE,
  HOUR,
  DAY,
}
```

rateCard:
```
/**
 * The type of the rate card.
 */
@friendlyName("RateCardType")
enum RateCardType {
  flatFee: "flat_fee",
  usageBased: "usage_based",
}
```

entitlement:
```
@friendlyName("EntitlementOrderBy")
enum EntitlementOrderBy {
  #suppress "@openmeter/api-spec/casing" "Use existing values"
  CreatedAt: "createdAt",
  #suppress "@openmeter/api-spec/casing" "Use existing values"
  UpdatedAt: "updatedAt",
}
```

For now I will just update the code to use the RateCard's format, but we should at least for new API's have some alignment here.

Reply:
Generally speaking given we are using lowerCamelCase for property names, I would recommend to use lowerCamelCase for values, but I would leave it for you folks.